diff -up gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fft_f64.c.1~ gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fft_f64.c
--- gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fft_f64.c.1~	2020-09-08 17:46:24.660311631 +0200
+++ gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fft_f64.c	2020-09-08 17:48:40.531667796 +0200
@@ -362,7 +362,7 @@ kiss_fft_f64_alloc (int nfft, int invers
 {
   kiss_fft_f64_cfg st = NULL;
   size_t memneeded = sizeof (struct kiss_fft_f64_state)
-      + sizeof (kiss_fft_f64_cpx) * (nfft - 1); /* twiddle factors */
+      + sizeof (kiss_fft_f64_cpx) * nfft; /* twiddle factors */
 
   if (lenmem == NULL) {
     st = (kiss_fft_f64_cfg) KISS_FFT_F64_MALLOC (memneeded);
diff -up gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fft_f64.h.1~ gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fft_f64.h
--- gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fft_f64.h.1~	2020-09-08 01:03:35.675583400 +0200
+++ gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fft_f64.h	2020-09-08 17:46:24.660311631 +0200
@@ -40,7 +40,7 @@ extern "C" {
 typedef struct {
     kiss_fft_f64_scalar r;
     kiss_fft_f64_scalar i;
-}kiss_fft_f64_cpx;
+} __attribute__((packed)) kiss_fft_f64_cpx;
 
 typedef struct kiss_fft_f64_state* kiss_fft_f64_cfg;
 
diff -up gst-plugins-base-1.18.0/gst-libs/gst/fft/_kiss_fft_guts_f64.h.1~ gst-plugins-base-1.18.0/gst-libs/gst/fft/_kiss_fft_guts_f64.h
--- gst-plugins-base-1.18.0/gst-libs/gst/fft/_kiss_fft_guts_f64.h.1~	2020-09-08 01:03:35.671583400 +0200
+++ gst-plugins-base-1.18.0/gst-libs/gst/fft/_kiss_fft_guts_f64.h	2020-09-08 17:46:24.660311631 +0200
@@ -37,8 +37,8 @@ struct kiss_fft_f64_state{
     int nfft;
     int inverse;
     int factors[2*MAXFACTORS];
-    kiss_fft_f64_cpx twiddles[1];
-};
+    kiss_fft_f64_cpx twiddles[0];
+} __attribute__((packed));
 
 /*
   Explanation of macros dealing with complex math:
diff -up gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fftr_f64.c.1~ gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fftr_f64.c
--- gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fftr_f64.c.1~	2020-09-08 17:46:24.660311631 +0200
+++ gst-plugins-base-1.18.0/gst-libs/gst/fft/kiss_fftr_f64.c	2020-09-08 17:48:08.180118845 +0200
@@ -17,7 +17,7 @@ struct kiss_fftr_f64_state
 #ifdef USE_SIMD
   void *pad;
 #endif
-};
+} __attribute__((packed));
 
 kiss_fftr_f64_cfg
 kiss_fftr_f64_alloc (int nfft, int inverse_fft, void *mem, size_t * lenmem)
