diff -uraN gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fft_f64.c omv-gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fft_f64.c
--- gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fft_f64.c	2024-03-05 00:51:42.000000000 +0100
+++ omv-gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fft_f64.c	2024-03-06 17:18:52.994567934 +0100
@@ -362,7 +362,7 @@
 {
   kiss_fft_f64_cfg st = NULL;
   size_t memneeded = sizeof (struct kiss_fft_f64_state)
-      + sizeof (kiss_fft_f64_cpx) * (nfft - 1); /* twiddle factors */
+      + sizeof (kiss_fft_f64_cpx) * nfft; /* twiddle factors */
 
   if (lenmem == NULL) {
     st = (kiss_fft_f64_cfg) KISS_FFT_F64_MALLOC (memneeded);
diff -uraN gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fft_f64.h omv-gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fft_f64.h
--- gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fft_f64.h	2024-03-05 00:51:42.000000000 +0100
+++ omv-gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fft_f64.h	2024-03-06 17:20:08.685599039 +0100
@@ -40,7 +40,7 @@
 typedef struct {
     kiss_fft_f64_scalar r;
     kiss_fft_f64_scalar i;
-}kiss_fft_f64_cpx;
+} __attribute__((packed)) kiss_fft_f64_cpx;
 
 typedef struct kiss_fft_f64_state* kiss_fft_f64_cfg;
 
diff -uraN gst-plugins-base-1.24.0/gst-libs/gst/fft/_kiss_fft_guts_f64.h omv-gst-plugins-base-1.24.0/gst-libs/gst/fft/_kiss_fft_guts_f64.h
--- gst-plugins-base-1.24.0/gst-libs/gst/fft/_kiss_fft_guts_f64.h	2024-03-05 00:51:42.000000000 +0100
+++ omv-gst-plugins-base-1.24.0/gst-libs/gst/fft/_kiss_fft_guts_f64.h	2024-03-06 17:21:24.320951167 +0100
@@ -37,8 +37,8 @@
     int nfft;
     int inverse;
     int factors[2*MAXFACTORS];
-    kiss_fft_f64_cpx twiddles[1];
-};
+    kiss_fft_f64_cpx twiddles[0];
+} __attribute__((packed));
 
 /*
   Explanation of macros dealing with complex math:
diff -uraN gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fftr_f64.c omv-gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fftr_f64.c
--- gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fftr_f64.c	2024-03-05 00:51:42.000000000 +0100
+++ omv-gst-plugins-base-1.24.0/gst-libs/gst/fft/kiss_fftr_f64.c	2024-03-06 17:22:59.003457851 +0100
@@ -17,7 +17,7 @@
 #ifdef USE_SIMD
   void *pad;
 #endif
-};
+} __attribute__((packed));
 
 kiss_fftr_f64_cfg
 kiss_fftr_f64_alloc (int nfft, int inverse_fft, void *mem, size_t *lenmem)
